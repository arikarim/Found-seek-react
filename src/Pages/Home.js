import axios from "axios";
import React, { useEffect } from "react";
import { Card, Col, Container, Row } from "react-bootstrap";

const Home = ({ devices, setDevices }) => {
  const fetchDevices = async () => {
    try {
      const data = await axios.get("http://localhost:3001/devices");
      setDevices(data.data);
    } catch (e) {
      console.log(e);
    }
  };
  useEffect(() => {
    fetchDevices();
    // eslint-disable-next-line
  }, []);

  return (
    <Container>
      <h1>Welcome to the home page</h1>
      <h3>Checkout The devices</h3>
      <Row className="d-flex flex-wrap">
        {devices &&
          devices.map((device) => (
            <Col key={device.id} className="mx-auto my-2" md={6} lg={4}>
              <Card className="mx-auto" style={{ width: "18rem", minHeight: "30rem" }}>
                <Card.Img
                  style={{ width: "18rem", height: "12rem" }}
                  variant="top"
                  src={device.images[0].url}
                />
                {/* {console.log(device)} */}
                <Card.Body>
                  <Card.Title>{device.name}</Card.Title>
                  <Card.Text>{device.description}</Card.Text>
                  <Card.Text className="position-absolute bottom-0">
                    Available:{" "}
                    {device.availability ? "Available" : "Not available"}
                  </Card.Text>
                </Card.Body>
              </Card>
            </Col>
          ))}
      </Row>
    </Container>
  );
};

export default Home;
