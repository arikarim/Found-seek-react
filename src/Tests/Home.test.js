import { render, screen } from "@testing-library/react";
import React from "react";
import { BrowserRouter, MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Home from "../Pages/Home";

describe("Header with Link", () => {
  test("full app rendering/navigating", () => {
    render(<Home />, { wrapper: MemoryRouter });
    // eslint-disable-next-line no-unused-expressions
    expect(screen.getByText("Welcome to the home page")).toBeInTheDocument;
  });
  test("should match with snapshot", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
