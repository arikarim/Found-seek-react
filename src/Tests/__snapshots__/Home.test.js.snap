// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header with Link should match with snapshot 1`] = `
<div
  className="container"
>
  <h1>
    Welcome to the home page
  </h1>
  <h3>
    Checkout The devices
  </h3>
  <div
    className="d-flex flex-wrap row"
  />
</div>
`;
